### PR TITLE
[WIP] Bayesrules

### PR DESCRIPTION
**Pull request by cast42**
# Thank you for opening a pull request!

Please check our [style guide](https://docs.pymc.io/en/latest/contributing/jupyter_style.html), and also make sure that the notebooks you've modified pass the `pre-commit` checks. If, for example, you modified `notebook1.ipynb` and `notebook2.ipynb`, you could do this by running:

```bash
pre-commit run --files notebook1.ipynb notebook2.ipynb
```

You may need to run it a couple of times before all the checks pass.

### CODE REVIEW

**Comment by canyon289** [comment]
Thanks for this PR! You caught us right before we had everything quite setup.

I merged the PR below, which adds a much stricter pymc environment and the Bayes_Rules directory. When you get a chance could you rebase this PR on at that one, and ensure the notebook runs with environment (which uses PyMC and python 3.9)?

https://github.com/pymc-devs/resources/pull/176

File: README.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ PyMC3 educational resources, including the PyMC3 port of the following books (or
 -  [PyMC3 port of the book "Bayesian Cognitive Modeling" by Michael Lee and EJ Wagenmakers](https://github.com/pymc-devs/resources/tree/main/BCM)
 -  [PyMC3 port of the book "Bayesian Statistical Methods" by Brian J. Reich and Sujit K. Ghosh](https://github.com/pymc-devs/resources/tree/main/BSM)
 -  [PyMC3 port of the book "Bayesian Data Analysis" by Andrew Gelman, John B. Carlin, Hal S. Stern, David B. Dunson, Aki Vehtari and Donald B. Rubin](https://github.com/pymc-devs/resources/tree/main/BDA3)
+-  [PyMC3/4? port of the book "Bayes Rules! An Introduction to Applied Bayesian Modeling" by Bayes Rules! An Introduction to Applied Bayesian Modeling
```
Comment (canyon289) [inline README.md:10]:
```suggestion
-  [PyMC port of the book "Bayes Rules! An Introduction to Applied Bayesian Modeling" by Bayes Rules! An Introduction to Applied Bayesian Modeling
```

File: bayesrules/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Bayes Rules! with Python and PyMC3/4
+
+[PyMC3/4? port of the book "Bayes Rules! An Introduction to Applied Bayesian Modeling" by Bayes Rules! An Introduction to Applied Bayesian Modeling
```
Comment (canyon289) [inline bayesrules/README.md:3]:
```suggestion
[PyMC port of the book "Bayes Rules! An Introduction to Applied Bayesian Modeling" by Bayes Rules! An Introduction to Applied Bayesian Modeling
```

File: bayesrules/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Bayes Rules! with Python and PyMC3/4
```
Comment (canyon289) [inline bayesrules/README.md:1]:
```suggestion
# Bayes Rules! with Python and PyMC
```

File: bayesrules/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+to install all the dependencies into an isolated environment. 
+
+Activate the environment by running:
+
+    source activate bayesrules
+
+To use the notebooks you first have to register your new environment as a valid notebook kernel:
+
+    python -m ipykernel install --user --name bayesrules --display-name "Python 3.10 (bayesrules)"
```
Comment (canyon289) [inline bayesrules/README.md:28]:
```suggestion
    python -m ipykernel install --user --name bayesrules --display-name "Python 3.9 (bayes_rules)"
```

File: bayesrules/environment.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+name: bayesrules
```
Comment (canyon289) [inline bayesrules/environment.yml:1]: Added a much stricter environment though I see I need to add pyreadr and watermark to it. I'll make the update

File: bayesrules/Section_2_1_Fake_news.ipynb [hunk 6]
```diff
@@ -0,0 +1,1517 @@
+{
```
Comment (canyon289) [inline bayesrules/Section_2_1_Fake_news.ipynb:1]: <p>Thanks! This note book is a good start though I'm realizing I should create a template to ensure each chapter is consistent. I will get to that in the next couple of days!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/resources/pull/179/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (canyon289) [reply bayesrules/Section_2_1_Fake_news.ipynb:1]: Thinking about it lets get this merged and then set it up to be a template afterwards. Should make things quite easy!

File: bayesrules/environment.yml (hunk 5 above)
Comment (cast42) [reply bayesrules/environment.yml:1]: I also need wget or request to download the R data files. Could you include one of them ?

Comment (cast42) [reply bayesrules/environment.yml:1]: Also I can't create this environment:
(base) lode@datarootss-MacBook-Pro-2 resources % cd Bayes_Rules/  
(base) lode@datarootss-MacBook-Pro-2 Bayes_Rules % conda env create -f environment.yml 
Collecting package metadata (repodata.json): done
Solving environment: failed

ResolvePackageNotFound: 
  - libzip==1.5.1=h8d318fa_1003
  - hdf4==4.2.13=h3ca952b_2
  - sniffio==1.2.0=py39h06a4308_1
  - matplotlib==3.5.1=py39h06a4308_1
  - jedi==0.18.1=py39h06a4308_1
  - numexpr==2.8.1=py39h6abb31d_0

Comment (canyon289) [reply bayesrules/environment.yml:1]: Apologies on that, ill get this fixed in next 24 hours

Comment (canyon289) [reply bayesrules/environment.yml:1]: When you get a chance can you try again? https://github.com/pymc-devs/resources/pull/181/files

Comment (cast42) [reply bayesrules/environment.yml:1]: I tried but again i get ResolvePackageNotFound....

Comment (canyon289) [reply bayesrules/environment.yml:1]: oh man, im sorry, environment setup is the worst. I hate to hold you up. Lets do this we can merge as is and then I'll figure out the environment outside of this PR. Thanks for being with me. I can get back to a review on this in the next day

File: bayesrules/Section_2_1_Fake_news.ipynb (hunk 6 above)
Comment (canyon289) [reply bayesrules/Section_2_1_Fake_news.ipynb:1]: Sounds great

**Comment by canyon289** [comment]
@cast42 any other changes you want to make? If not I'll merge this as is and in another PR fix up the environment, likely make some minor changes, all to set the template for all chapters moving forward

**Comment by cast42** [comment]
@canyon289  I removed all the wget code. Fine for me to merge now. Thx for your help.

File: bayesrules/environment.yml (hunk 5 above)
Comment (cast42) [reply bayesrules/environment.yml:1]: No problem. Agree to proceed as you propose. I'm sure to resolve the environment issues. I use environment_simple.yml for the moment, so no holding up.

**Comment by canyon289** [comment]
> @canyon289 I removed all the wget code. Fine for me to merge now. Thx for your help.

Thanks! Can you run precommit to get the notebook up to standard and then I'll merge (as a matter of good practice)

**Comment by cast42** [comment]
Note very familiear with pre-commit but Chapter[2|3]*.ipynb should be fine now. Let me know if I missed something.

File: .gitignore [hunk 7]
```diff
@@ -99,3 +99,11 @@ ENV/
 
 # mypy
 .mypy_cache/
+
+.DS_Store
+*.rda
+.vscode/settings.json
+bayesrules/Section_2_1_Fake_news.ipynb
+BCM/*
```
Comment (canyon289) [inline .gitignore:107]: Why are all these added to gitignore?

Comment (cast42) [reply .gitignore:107]: To prevent that they appear in the pull request.

**Comment by canyon289** [comment]
Closing as https://github.com/pymc-devs/resources/pull/184 captures all the changes

**Pull request closed without merging**